### DISCUSSION TITLE
Permalink actived --> Internal Server Error with nginx reverse proxy

### DISCUSSION THREAD

**Discussion by cpc6128** [docker-library/wordpress#652] [General]
Hi,
Everything works fine but when I activate the permalink option in wordpress, I get this error:

```
Internal Server Error
The server encountered an internal error or misconfiguration and was unable to complete your request.
Please contact the server administrator at webmaster@localhost to inform them of the time this error occurred, and the actions you performed just before this error.

More information about this error may be available in the server error log.

Apache/2.4.38 (Debian) Server at www.corunning.fr Port 80
```

My Nginx conf :

```
server {

    listen 443 ssl http2;
    listen [::]:443 ssl http2;

......

   location ^~ /blog/ {

        rewrite /blog/(.*) /$1  break;
        #rewrite /blog / break;

        proxy_pass    http://wordpress;
        proxy_redirect   off;

        proxy_read_timeout    90;
        proxy_connect_timeout 90;

        proxy_set_header      Host $host;
        proxy_set_header      X-Real-IP $remote_addr;
        proxy_set_header      X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header      X-Forwarded-Host $server_name;
        proxy_set_header      X-Forwarded-Proto https;
    }
```

My docker-compose conf : 

```
 wordpress:
     depends_on:
       - dbwp
     image: wordpress:5.5.1
     restart: always
     volumes:
         - wp_data:/var/www/html/wp-content:rw
     environment:
       WORDPRESS_DB_HOST: dbwp:3306
       WORDPRESS_DB_USER: xx
       WORDPRESS_DB_PASSWORD: xx
       WORDPRESS_CONFIG_EXTRA: |
        define( 'WP_SITEURL', 'https://www.corunning.fr/blog' );
        define( 'WP_HOME', 'https://www.corunning.fr/blog' );
        define( 'TEST_COOKIE', 'testezmoi' );
        $$_SERVER['REQUEST_URI'] = str_replace("/wp-admin/", "/blog/wp-admin/",  $$_SERVER['REQUES$

```

Error in docker : 

```
wordpress_1  | 88.27.32.70 - - [08/Oct/2021:10:18:00 +0000] "GET /2021/09/16/choisir-sa-paire-de-running-connaitre-son-type-de-foul/ HTTP/1.0" 500 806 "https://www.corunning.fr/blog/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.71 Safari/537.36"
wordpress_1  | [Fri Oct 08 10:18:00.468440 2021] [core:error] [pid 83] [client 88.27.32.70:0] AH00124: Request exceeded the limit of 10 internal redirects due to probable configuration error. Use 'LimitInternalRecursion' to increase the limit if necessary. Use 'LogLevel debug' to get a backtrace., referer: https://www.corunning.fr/blog/
```

Any idea on what I forgot to configure?
Thank you !

Ronan

**Comment by tianon** [comment]
I think https://github.com/docker-library/wordpress/issues/370#issuecomment-459138751 might be helpful for you.

TLDR, I'd suggest ditching all the `WP_SITEURL`, `WP_HOME`, `REQUEST_URI` bits and instead setting your workdir to a subdirectory:

```yaml
 wordpress:
     depends_on:
       - dbwp
     image: wordpress:5.5.1
     restart: always
     working_dir: /var/www/html/blog
     volumes:
         - wp_data:/var/www/html/blog/wp-content:rw
```

**Reply by cpc6128** [reply]
thank you so much @tianon . This is the solution to my problem!
